app/screens/edit_profile/components/profile_image_picker.tsx

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {useIntl} from 'react-intl';
+import {TouchableOpacity} from 'react-native';
+
+import {Client} from '@client/rest';
+import CompassIcon from '@components/compass_icon';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import NetworkManager from '@init/network_manager';
+import PanelItem from '@screens/edit_profile/components/panel_item';
+import {bottomSheet} from '@screens/navigation';
+import PickerUtil from '@utils/file/file_picker';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import type UserModel from '@typings/database/models/servers/user';
+import type {UploadExtractedFile} from '@typings/utils/file';
+
+const hitSlop = {top: 100, bottom: 20, right: 20, left: 100};
+const ACTION_HEIGHT = 55;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        touchable: {
+            alignItems: 'center',
+            justifyContent: 'center',
+            borderWidth: 1,
+            borderColor: theme.centerChannelBg,
+            borderRadius: 18,
+            height: 36,
+            width: 36,
+            position: 'absolute',
+            bottom: 0,
+            right: 0,
+            backgroundColor: theme.centerChannelBg,
+        },
+    };
+});
+
+type ImagePickerProps = {
+    onRemoveProfileImage: UploadExtractedFile;
+    uploadFiles: UploadExtractedFile;
+    user: UserModel;
+};
+
+const hasPictureUrl = (user: UserModel, serverUrl: string) => {
+    const {id, lastPictureUpdate} = user;
+
+    let client: Client | undefined;
+    let profileImageUrl: string | undefined;
+
+    try {
+        client = NetworkManager.getClient(serverUrl);
+        profileImageUrl = client.getProfilePictureUrl(id, lastPictureUpdate);
+    } catch {
+        return false;
+    }
+
+    // Check if image url includes query string for timestamp. If so, it means the image has been updated from the default, i.e. '.../image?_=1544159746868'
+    return Boolean(profileImageUrl?.includes('image?_'));
+};
+
+const ProfileImagePicker = ({
+    onRemoveProfileImage,
+    uploadFiles,
+    user,
+}: ImagePickerProps) => {
+    const theme = useTheme();
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const pictureUtils = useMemo(() => new PickerUtil(intl, uploadFiles), [uploadFiles]);
```

Comment:
@larkox I guess this is being memoized to avoid having to create a new PickerUtil object every time this component re-renders, which in my mind makes total sense